internal/cmd/server.go

```diff
@@ -95,3 +90,6 @@ func defaultServerOptions() server.Options {
 var runServer = func(ctx context.Context, srv *server.Server) error {
 	return srv.Run(ctx)
 }
+
+// newServer is a shim for testing.
+var newServer = server.New
```

Comment:
What's the overhead of this? Does it mean instantiating 2 servers every time we want to run one?

Reply:
There shouldn't be any extra overhead (definitely not creating two servers).

This line is capturing a reference to the `server.New` function in the variable `newServer`. Functionally it would be similar to this:

```go
func newServer(opts server.Options) (*Server, error) {
    return server.New(opts)
}
```

The only difference is with `var newServer = server.New` we have one less function call. Calling `newServer(...)` is exactly the same as calling `server.New(...)`, it's the same function, just captured to a different variable.

Does that make sense? I might have misunderstood your question

Reply:
That makes sense, thanks